src/lib/Types/Core.hs

```diff
@@ -2140,6 +2148,12 @@ instance ExtOutMap Env TopEnvFrag where
 
       lookupModulePure v = case lookupEnvPure (Env newTopEnv mempty) v of ModuleBinding m -> m
 
+addMethods :: (SpecDictName n, [NaryLamExpr n]) -> Env n -> Env n
+addMethods (dName, methods) e = do
+  let SpecializedDictBinding (SpecializedDict dAbs _) = lookupEnvPure e dName
```

Comment:
Might be a good idea to pattern match on `Nothing` to make sure we don't add them multiple times.